validator/src/journal/candidate_block.rs

```diff
@@ -431,10 +430,17 @@ impl CandidateBlock {
                     committed_txn_cache.add_batch(&batch.clone());
                 }
             } else {
+                self.invalid_batch_ids.insert(batch.header_signature.clone());
```

Comment:
Looks like line 413 to line 415 intended to remove bad batches from the pending batches?